src/compute-client/src/plan/mod.rs

```diff
@@ -590,9 +594,22 @@ impl RustType<ProtoPlan> for Plan {
                     body: Some(body.into_proto()),
                 }
                 .into()),
-                Plan::LetRec { ids, values, body } => LetRec(
+                Plan::LetRec {
+                    ids,
+                    values,
+                    max_iters,
+                    body,
+                } => LetRec(
                     ProtoPlanLetRec {
                         ids: ids.into_proto(),
+                        max_iters: max_iters
+                            .into_iter()
+                            .map(|d| match d {
```

Comment:
minor nit: can you add this 

```rust
impl RustType<u64> for Option<NonZeroU64> {
    fn into_proto(&self) -> u64 {
        match self {
            Some(d) => d.get(),
            None => 0,
        }
    }

    fn from_proto(proto: u64) -> Result<Self, TryFromProtoError> {
        Ok(NonZeroU64::new(proto))
    }
}
```

after this line

```rust
rust_type_id![bool, f32, f64, i32, i64, String, u32, u64, Vec<u8>];
```

and then use `max_iters.into_proto()` and `proto.max_iters.into_rust()?`?